.github/workflows/merge-group.yml

```diff
@@ -0,0 +1,134 @@
+      - uses: Swatinem/rust-cache@v2
+      - name: Linux Tests
+        run: |
+          cargo nextest run --profile ci --workspace --cargo-profile dev-ci
+      - name: Linux Gadget Tests w/o debug assertions
+        run: |
+          cargo nextest run --profile ci --workspace --cargo-profile dev-no-assertions -E 'test(circuit::gadgets)'
+
+  mac-m1:
+    if: github.event_name != 'pull_request' || github.event.action == 'enqueued'
+    runs-on: macos-latest-xlarge
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions-rs/toolchain@v1
+      - uses: taiki-e/install-action@nextest
+      - uses: Swatinem/rust-cache@v2
+      - name: Linux Tests
+        run: |
+          cargo nextest run --profile ci --workspace --cargo-profile dev-ci
+      - name: Linux Gadget Tests w/o debug assertions
+        run: |
+          cargo nextest run --profile ci --workspace --cargo-profile dev-no-assertions -E 'test(circuit::gadgets)'
+
+  # TODO: Make this a required status check
+  # TODO: Cache successful bench run from PR branch on master, keyed on commit hash
+  # Run comparative benchmark against master
+  gpu-benchmark:
+    # [TEMPORARY] Test one run before attempting merge
+    #if: github.event_name != 'pull_request' || github.event.action == 'enqueued'
+    name: Run fibonacci bench on GPU
+    runs-on: [self-hosted, gpu-bench-t4]
+    steps:
+      # TODO: Factor this out into an action or into justfile, it's used in 4 places
+      # Set up GPU
+      # Check we have access to the machine's Nvidia drivers
+      - run: nvidia-smi
+      # Check that CUDA is installed with a driver-compatible version
+      # This must also be compatible with the GPU architecture, see above link
+      - run: nvcc --version
+      - uses: actions/checkout@v4
+      # Checkout base branch for comparative bench
+      - uses: actions/checkout@v4
+        with:
+          ref: master
+      - run: ls -a
+      - name: Set base ref variable
+        run: echo "BASE_REF=$(git rev-parse HEAD)" >> $GITHUB_ENV
+      # Checkout the justfile and env of the source branch so the base can bench
+      - run: git restore --source ${{ github.sha }} justfile bench.env
+      - run: ls -a
+      # Install dependencies
+      - uses: actions-rs/toolchain@v1
+      - uses: Swatinem/rust-cache@v2
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: just@1
+      # Run benchmark on base branch
+      - name: Install criterion
+        run: |
+          cargo install cargo-criterion
+          cargo install criterion-table
+      - name: Run GPU bench
+        run: just --dotenv-filename bench.env gpu-bench fibonacci
+      # Switch to triggering branch and run benchmark
+      - run: rm justfile bench.env
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.sha }}
+      - name: Run GPU bench on source branch
+        run: just --dotenv-filename bench.env gpu-bench fibonacci
```

Comment:
We should eventually edit this job to *not* run the bench on the triggering branch, but instead retrieve the benchmark values from the bench results for the same machine, hosted in github-pages. Those bench results can then be fed to `criterion-table` as-is.
This does not have to be part of this PR, but we should open an issue for this optimization.

Reply:
See #812